MobiFlight/MobiFlightCache.cs

```diff
@@ -128,91 +121,47 @@ public IEnumerable<MobiFlightModule> GetModules()
                 RegistryKey regUSB = regLocalMachine.OpenSubKey(regPath);
                 if (regUSB == null) continue;
 
-                String[] arduinoVidPids = MobiFlightModuleInfo.VIDPID_MICRO
-                                            .Concat(MobiFlightModuleInfo.VIDPID_MEGA)
-                                            .Concat(MobiFlightModuleInfo.VIDPID_UNO).ToArray();
-
-                Regex regEx = new Regex("^(" + string.Join("|", arduinoVidPids) + ")");
-
-                Log.Instance.log(regEx.ToString(), LogSeverity.Debug);
-
                 foreach (String regDevice in regUSB.GetSubKeyNames())
                 {
                     String message = null;
-                    Log.Instance.log("Checking for compatible module: " + regDevice, LogSeverity.Debug);
+                    Log.Instance.log($"Checking for compatible module: {regDevice}", LogSeverity.Debug);
 
                     foreach (String regSubDevice in regUSB.OpenSubKey(regDevice).GetSubKeyNames())
                     {
+                        Board board;
                         String FriendlyName = regUSB.OpenSubKey(regDevice).OpenSubKey(regSubDevice).GetValue("FriendlyName") as String;
                         if (FriendlyName == null) continue;
 
-                        // determine type based on names
-                        if (FriendlyName.Contains("Mega 2560"))
+                        // Check to see if any board configurations exist
+                        board = BoardDefinitions.GetBoardByFriendlyName(FriendlyName) ?? BoardDefinitions.GetBoardByHardwareId(regDevice);

```

Comment:
All of the previous checks by vid/pid and friendly name condense down to this single line. Cool :)